File: Chess/src/main/java/side_server/ClientListenThread.java
package side_server;

import msg.Message;

import java.io.IOException;
import java.util.logging.Level;
import java.util.logging.Logger;

public class ClientListenThread extends Thread {
    public SClient client;

    public ClientListenThread(SClient client) {
        this.client = client;
    }

    @Override
    public void run() {
        while (!this.client.socket.isClosed()) {
            try {
                Message msg = (Message) (this.client.cInput.readObject());
                switch (msg.type) {
                    case PAIRING:
                        this.client.isWantToPair = true;
                        this.client.pairingThread.start();
                        break;
                    case MOVE:
                    case CHECK:
                        this.client.pair.send(msg);
                        break;
                    case END:
                        this.client.pair = null;
                    case LEAVE:
                        this.client.isPaired = false;
                        this.client.isWantToPair = false;
                        assert this.client.pair != null;
                        this.client.pair.isWantToPair = false;
                        this.client.pair.isPaired = false;
                        this.client.pair.pair = null;
                        this.client.pair = null;

                }
            } catch (IOException | ClassNotFoundException ex) {
                Logger.getLogger(ClientListenThread.class.getName()).log(Level.SEVERE, null, ex);
            }
        }
    }
}

File: Chess/src/main/kotlin/hummel/side_server/ClientListenThread.kt
package main.kotlin.hummel.side_server

import main.kotlin.hummel.msg.Message
import main.kotlin.hummel.msg.Message.MessageTypes
import java.io.IOException
import java.util.logging.Level
import java.util.logging.Logger

class ClientListenThread(var client: SClient) : Thread() {
	override fun run() {
		while (!client.socket!!.isClosed) {
			try {
				val msg = client.cInput!!.readObject() as Message
				when (msg.type) {
					MessageTypes.PAIRING -> {
						client.isWantToPair = true
						client.pairingThread!!.start()
					}

					MessageTypes.MOVE, MessageTypes.CHECK -> client.pair!!.send(msg)
					MessageTypes.END -> {
						client.pair = null
						client.isPaired = false
						client.isWantToPair = false
						client.pair = null
					}
					MessageTypes.LEAVE -> {
						client.isPaired = false
						client.isWantToPair = false
						client.pair!!.isWantToPair = false
						client.pair!!.isPaired = false
						client.pair!!.pair = null
						client.pair = null
					}

					else -> {}
				}
			} catch (ex: IOException) {
				Logger.getLogger(ClientListenThread::class.java.name).log(Level.SEVERE, null, ex)
			} catch (ex: ClassNotFoundException) {
				Logger.getLogger(ClientListenThread::class.java.name).log(Level.SEVERE, null, ex)
			}
		}
	}
}